kolibri/core/assets/src/state/store.js

```diff
@@ -139,9 +139,16 @@ export const coreMutations = {
     state.core.logging.attempt.interaction_history.push(action);
   },
   UPDATE_LOGGING_MASTERY(state, currentTime, correct, firstAttempt, hinted) {
+    // Check interaction history for any errors
+    let interactionHistory = state.core.logging.attempt.interaction_history;
```

Comment:
I think we can get rid of this, and just do the smaller fix of just setting the question to incorrect, and saving the attempt.

Reply:
@rtibbles my purpose for this one was to determine whether to add those tiny check or X marks when the exam failed to load after hitting next question.

`state.core.logging.mastery.pastattempts.unshift({ correct, hinted });`

Is there any other way to prevent it from adding those to the past attempts?

Reply:
The issue for me of trying to hide these past attempts is two fold:
* If we don't record the attempt, then if you refresh the page, you are returned to the item that has the error.
* If we do record the attempt, but just hide it from the user with this check, then it is not displayed to the user but all other logic continues to act as if this attempt happened.

The reason the first of these is particularly concerning is not so much that it will return you to the error question, but that when you return to the exercise (having gone past the question), you will be returned to the *wrong* question. Hence, we should just mark the item incorrect, record it, move on, and be transparent with the user that that is what is happening. The reason that I had made this 'correct' in the past was to stop completion of an exercise being prevented due to a faulty question, but apparently, this is not what some users want!

Reply:
I always get confused about `MasteryLog` vs `InteractionHistory`. Looks like `InteractionHistory` supports `InteractionTypes.error` while `MasteryLog` does not?

Could we add 'error' as a possible MasteryLog outcome? Then we can log them but treat them differently?

Reply:
> Doesn't that unjustly penalize the user for something they have no control over? Imagine if they had a streak, and then it was broken because an exercise couldn't load...

I can see some learners complaining that they wouldn't get a streak on their first try because of some faulty question.

Imagine there are 8 questions and then one of them has a problem. One learner gets a streak on their first try and the other gets their streak broken because their was a problem loading it


> Yes, that is why the behaviour is the way it is currently, but the issue said that this was a problem.

I also wouldn't like to get a check mark because of a faulty question, but also, I don't want to break my streak because of a faulty question.

Anyways, I'll make the requested changes

Reply:
Yeah, I expect this might replace one issue with its inverse

Reply:
maybe it's a slight improvement as it will in the end force more learning rather than less :)

Reply:
MasteryLog is a summary of mastery for an exercise, based on a particular mastery criterion. InteractionHistory is the history of interaction with a single attempt at a single question (the interaction history is a JSON blob on the attemptlog).

We could add a field to the attemptlog to indicate that an error occurred, or to have some 'type' 'answer given' 'hint given' 'question error', although it might be better as a boolean for 'errored' if we wanted to efficiently filter by it (and there is already a 'hinted' boolean flag).